files/external_node_v2.rb

```diff
@@ -196,8 +196,12 @@ def enc(certname)
   end
   res = http.start { |http| http.request(req) }
 
-  raise "Error retrieving node #{certname}: #{res.class}\nCheck Foreman's /var/log/foreman/production.log for more information." unless res.code == "200"
-  res.body
+  return res.body if res.code == '200'
+
+  raise Net::HTTPServerException if (500..599).include?(res.code)
```

Comment:
This doesn't work for me, as `res.code` is a string but it's being compared against an integer range.